compiler/rustc_ast/src/ast.rs

```diff
@@ -2440,7 +2440,7 @@ pub struct ForeignMod {
 
 #[derive(Clone, Encodable, Decodable, Debug)]
 pub struct EnumDef {
-    pub variants: Vec<Variant>,
+    pub variants: Box<[Variant]>,
```

Comment:
Maybe we could use `std:.boxed::ThinBox` or something similar to further reduce the size from `16` to `8` bytes? By storing the slice size on the heap.

Reply:
There is also `thin_vec::ThinVec`, which I introduced to the compiler in #100869. Unfortunately, getting any of these kinds of changes to show a clear performance benefit has been hard.